cmd/tempo/app/modules.go

```diff
@@ -436,6 +437,21 @@ func (t *App) initCompactor() (services.Service, error) {
 	return t.compactor, nil
 }
 
+func (t *App) initOptionalStore() (services.Service, error) {
+	// Used by the local-blocs processor to flush RF1 blocks to storage.
+	// Only initialize if it's configured.
+	if t.cfg.StorageConfig.Trace.Backend == "" {
+
+		if t.cfg.Target == MetricsGenerator && t.cfg.Generator.Processor.LocalBlocks.FlushToStorage {
```

Comment:
nit: Can we move this check to initGenerator?

Reply:
Good point, done